java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -286,13 +286,13 @@ module ModelValidation {
       not kind.matches("qltest%") and
       result = "Invalid kind \"" + kind + "\" in sink model."
     )
-    // All source kinds are supported, but some may not be used in this query.
-    // or
-    // exists(string kind | sourceModel(_, _, _, _, _, _, _, kind, _) |
-    //   not ConfiguredExtensions::supportedSourceKind(kind) and
-    //   not kind.matches("qltest%") and
-    //   result = "Invalid kind \"" + kind + "\" in source model."
-    // )
+    or
+    exists(string kind | sourceModel(_, _, _, _, _, _, _, kind, _) |
+      not kind =
+        ["remote", "sql", "contentprovider", "android-widget", "android-external-storage-dir"] and
```

Comment:
Does this mean that custom source kinds will fail?

Reply:
The `ModelValidation` module where this is defined contains the query predicate `invalidModelRow`. The actual validation check (where this query is executed) are the tests where the `ModelValidation` module is imported. The module is only imported in tests that has dependencies on `codeql/java-all` and `codeql/java-queries`. This means that it is only the models in `codeql/java-all` and `codeql/java-queries` and those that are included in the actual test that are validated.
If you make your own query pack then these models are not validated unless you write a specific test for it.